prExplorer/client-app/src/app/stores/categoryStore.ts

```diff
@@ -0,0 +1,129 @@
+import { makeAutoObservable, runInAction } from "mobx";
+import agent from "../api/agent";
+import { v4 as uuid } from "uuid";
+import { toast } from "react-toastify";
+import { Category } from "../models/category";
+
+export default class CategoryStore {
+  categoryRegistry = new Map<string, Category>();
+  selectedCategory: Category | undefined = undefined;
+  editMode = false;
+  loading = false;
+  loadingInitial = false;
+
+  constructor() {
+    makeAutoObservable(this);
+  }
+
+  get categoriesSortedByName() {
+    const categoryList = Array.from(this.categoryRegistry.values());
+    categoryList.sort((a, b) => {
+      const titleA = a.title.toLowerCase();
+      const titleB = b.title.toLowerCase();
+      return titleA.localeCompare(titleB);
+    });
+    return categoryList;
+  }
+
+  loadCategories = async () => {
+    this.setLoadingInitial(true);
+    try {
+      const categories = await agent.Categories.list();
+      categories.forEach((category) => {
+        this.setCategory(category);
+      });
+      this.setLoadingInitial(false);
+    } catch (error) {
+      console.log(error);
```

Comment:
`console.log` shouldn't be checked-in

Reply:
We'll create new controller on API side for error logs.(Front-Back). Use seperate db (mongoDb cloud) there.